test/chaos/src/main.rs

```diff
@@ -30,15 +31,86 @@ async fn run() -> Result<(), anyhow::Error> {
     let args = Args::from_args();
     env_logger::init();
 
+    match args.test {
+        Tests::BytesToKafka => bytes_to_kafka(args).await?,
+        Tests::MysqlDebeziumKafka => mysql_debezium_kafka(args).await?,
+    }
+
+    log::info!(
+        "Completed chaos testing in {} milliseconds",
+        timer.elapsed().as_millis()
+    );
+    Ok(())
+}
+
+async fn mysql_debezium_kafka(args: Args) -> Result<(), anyhow::Error> {
+    let run_seconds = args.run_seconds.unwrap_or(864_000);
+    let end = std::time::Instant::now().add(Duration::from_secs(run_seconds));
+    log::info!(
+        "starting chaos test {:#?} with mzd={}:{} kafka={} run seconds={}",
+        args.test,
+        args.materialized_host,
+        args.materialized_port,
+        args.kafka_url,
+        run_seconds,
+    );
+
+    let mz_client = mz_client::client(&args.materialized_host, args.materialized_port).await?;
+
+    // Create Kafka source.
+    let src_query = "CREATE SOURCE src_orderline
+                           FROM KAFKA BROKER 'kafka:9092' TOPIC 'mysql.tpcch.orderline'
+                           FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'http://schema-registry:8081'
+                           ENVELOPE DEBEZIUM;";
+    log::info!("creating source=> {}", src_query);
+    mz_client.execute(&*src_query, &[]).await?;
+
+    // Create count materialized view.
+    let count_query = "CREATE MATERIALIZED VIEW orderline_count AS
+                             SELECT count(*) FROM src_orderline;";
+    log::info!("creating view=> {}", count_query);
+    mz_client.execute(&*count_query, &[]).await?;
+
+    // Create Q01 materialized view.
+    let qo1_query = "CREATE MATERIALIZED VIEW q01 AS
+                           SELECT
+                               ol_number,
+                               sum(ol_quantity) AS sum_qty,
+                               sum(ol_amount) AS sum_amount,
+                               avg(ol_quantity) AS avg_qty,
+                               avg(ol_amount) AS avg_amount,
+                               count(*) AS count_order
+                           FROM src_orderline
+                           WHERE ol_delivery_d > TIMESTAMP '2007-01-02 00:00:00.000000'
+                           GROUP BY ol_number
+                           ORDER BY ol_number;";
+    log::info!("creating view=> {}", qo1_query);
+    mz_client.execute(&*qo1_query, &[]).await?;
+
+    // Query both materialized views until run_seconds has expired, raise any errors.
+    while std::time::Instant::now() < end {
+        let count = "SELECT * FROM orderline_count;";
+        mz_client::try_query_one(&mz_client, &*count, Duration::from_secs(1)).await?;
+
+        let q01 = "SELECT * FROM q01;";
+        mz_client::try_query(&mz_client, &q01, Duration::from_secs(1)).await?;
+    }
```

Comment:
this is good, but I think going forward this should probably share some of the validation code with checker. I think you've already said that you would like to share the code in `test_utils` or something, just calling it out.

Reply:
yeah agreed!